pkg/gce-pd-csi-driver/controller.go

```diff
@@ -78,29 +79,52 @@ func (gceCS *GCEControllerServer) CreateVolume(ctx context.Context, req *csi.Cre
 	// Apply Parameters (case-insensitive). We leave validation of
 	// the values to the cloud provider.
 	diskType := "pd-standard"
-	configuredZone := gceCS.CloudProvider.GetZone()
+	var configuredZone string
+
 	for k, v := range req.GetParameters() {
 		if k == "csiProvisionerSecretName" || k == "csiProvisionerSecretNamespace" {
 			// These are hardcoded secrets keys required to function but not needed by GCE PD
 			continue
 		}
 		switch strings.ToLower(k) {
-		case "type":
+		case common.ParameterKeyType:
 			glog.Infof("Setting type: %v", v)
 			diskType = v
-		case "zone":
+		case common.ParameterKeyZone:
 			configuredZone = v
 		default:
 			return nil, status.Error(codes.InvalidArgument, fmt.Sprintf("invalid option %q", k))
 		}
 	}
 
+	// Topology overwrites Storage Class Params if specified
```

Comment:
I think we should not allow both zone/zones and AllowedTopologies to be specified at the same time.  But I'm not sure if we can actually detect that because csi provisioner will always fill in the requisite topologies, even when AllowedTopologies is not specified. cc @verult 

So I think the next best option would be to use the zone/zones parameters because that's what the user specified, so we shouldn't just ignore it.